src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -2109,8 +2107,16 @@ JRT_LEAF(void, SharedRuntime::fixup_callers_callsite(Method* method, address cal
   // "to interpreter" stub in order to load up the Method*. Don't
   // ask me how I know this...
 
+  // Result from nmethod::is_unloading is not stable across safepoints.
+  NoSafepointVerifier nsv;
+
+  CompiledMethod* callee = moop->code();
```

Comment:
There is a moop->code() null check just a few lines below, so now it looks like we are reading the code pointer twice checking if it is null. Is ot enough to do that one time?

Reply:
It's actually the same number of null checks as before, if you look at what from_compiled_entry_no_trampoline() used to do.  But I did consider removing the 2nd check, because no matter how late we check, we can always lose the race where it becomes null right after our last check.  It's harmless however, so I decided to keep it.

Reply:
Okay. That seems fine.